crates/wac-graph/src/plug.rs

```diff
@@ -0,0 +1,64 @@
+use crate::{
+    types::{Package, SubtypeChecker},
+    CompositionGraph, EncodeOptions, NodeId, PackageId,
+};
+
+/// Take the exports of the plug component and plug them into the socket component.
+pub fn plug(
+    graph: &mut CompositionGraph,
+    plugs: Vec<Package>,
```

Comment:
I think we probably want this to be `Vec<PackageId>` and have the caller register the packages. That gives the caller the ability to do more things with the package without awkward work arounds like needing to use `CompositionGraph::get_package_by_name`.